packages/flutter_goldens/lib/flutter_goldens.dart

```diff
@@ -324,9 +328,12 @@ class FlutterPreSubmitFileComparator extends FlutterGoldenFileComparator {
   /// performed as pre-submit tests with Skia Gold.
   static bool isAvailableForEnvironment(Platform platform) {
     final String cirrusPR = platform.environment['CIRRUS_PR'] ?? '';
+    final bool hasWritePermission = platform.environment['CIRRUS_USER_PERMISSION'] == 'admin'
+      || platform.environment['CIRRUS_USER_PERMISSION'] == 'write';
```

Comment:
We could also check whether `platofrm.environment['GOLD_SERVICE_ACCOUNT'].startsWith('ENCRYPTED')`.  If so, we don't have a valid key.